apps/site/util/downloadUtils/archive.tsx

```diff
@@ -0,0 +1,164 @@
+  platform: UserPlatform,
+  version: string
+): boolean {
+  const {
+    os: osList,
+    platform: platformList,
+    semver: versions,
+  } = compatibility;
+
+  return (
+    (osList?.includes(os) ?? true) &&
+    (platformList?.includes(platform) ?? true) &&
+    (versions?.every(r => semVer.satisfies(version, r)) ?? true)
+  );
+}
+
+type CompatibleArtifactOptions = {
+  platforms?: Record<UserOS, Array<DownloadDropdownItem<UserPlatform>>>;
+  exclude?: Array<string>;
+  version: string;
+  kind?: DownloadKind;
+};
+
+/**
+ * Returns a list of compatible artifacts for the given options.
+ */
+const getCompatibleArtifacts = ({
+  platforms = PLATFORMS,
+  exclude = [],
+  version,
+  kind = 'binary',
+}: CompatibleArtifactOptions): Array<NodeDownloadArtifact> => {
+  return Object.entries(platforms).flatMap(([os, items]) => {
+    if (exclude.includes(os)) return [];
+
+    const operatingSystem = os as UserOS;
+
+    return items
+      .filter(({ compatibility, value }) =>
+        isCompatible(compatibility, operatingSystem, value, version)
+      )
+      .map(({ value, label }) => {
+        const url = getNodeDownloadUrl({
+          version: version,
+          os: operatingSystem,
+          platform: value,
+          kind: kind,
+        });
+
+        return {
+          file: url.replace(`${DIST_URL}${version}/`, ''),
+          kind: kind,
+          os: operatingSystem,
+          architecture: label,
+          url: url,
+          version: version,
+        };
+      });
+  });
+};
+
+/**
+ * Builds the release artifacts for a given Node.js release and version.
+ * It retrieves binaries, installers, and source files based on the version.
+ */
+export const buildReleaseArtifacts = (
+  release: NodeRelease,
+  version: string
+) => {
+  const minorVersion = release.minorVersions.find(
+    ({ versionWithPrefix }) => versionWithPrefix === version
+  );
+
+  const enrichedRelease = {
+    ...release,
+    ...minorVersion,
+  };
+
+  return {
+    binaries: getCompatibleArtifacts({
+      version: version,
+      kind: 'binary',
+    }),
+    installers: getCompatibleArtifacts({
+      exclude: OS_NOT_SUPPORTING_INSTALLERS,
+      version: version,
+      kind: 'installer',
+    }),
+    sources: {
+      shasum: getNodeDownloadUrl({
+        version: version,
+        kind: 'shasum',
+      }),
+      tarball: getNodeDownloadUrl({
+        version: version,
+        kind: 'source',
+      }),
+    },
+    version: version,
+    release: enrichedRelease,
+  };
+};
+
+/**
+ * Extracts the version from the pathname.
+ * It expects the version to be in the format 'v22.0.4' or 'archive'.
+ */
+export const extractVersionFromPath = (pathname: string | undefined) => {
+  if (!pathname) return null;
+
+  const segments = pathname.split('/').filter(Boolean);
+  const version = segments.pop();
+
+  // Check version format like (v22.0.4 or 'archive')
+  if (!version || (!version.match(/^v\d+(\.\d+)*$/) && version !== 'archive')) {
```

Comment:
nit: in `next.dynamic.constants.mjs`, you use `/^download\/(v\d+(\.\d+)*|archive)$/` as the regex, why not use the same thing here?